lib/internal/net.js

```diff
@@ -9,3 +9,9 @@ function isLegalPort(port) {
     return false;
   return +port === (port >>> 0) && port >= 0 && port <= 0xFFFF;
 }
+
+
+function assertPort(port) {
+  if (typeof port !== 'undefined' && !isLegalPort(port))
```

Comment:
Is the following a test?
```js
typeof net.createServer(()=>{}).listen().address().port === 'number';
```
If `undefined` is meant to be allowed to pass through then should probably document that at least in the form of a test.